Add StarTeam SCM processor

paul-murray-mf opened the pull request:
Please Make sure these boxes are checked before submitting your pull request - Thanks ahead!

- [ ] Proper pull request title - concise and clear for others.
- [ ] Proper pull request short description - clear and concise (as it should appear in the Jira ticket) for other developers and users.
- [ ] Proper Jira ticket - Number, Link in pull request description.
- [ ] The PR can is merged on your machine without any conflicts.
- [ ] The PR can is built on your machine without any (new) warnings.
- [ ] The PR passed sanity tests by you / QA / DevTest / Good Samaritan.
- [ ] Add unit tests with new features.
- [ ] If you added any dependency to the POM - Please update grount

paul-murray-mf committed:
Add StarTeam SCM processor

radislavB commented on pom.xml:
```diff
@@ -478,7 +478,7 @@
+			<version>2.0.71.3</version>
 		</dependency>
 
 		<!--BUILDER providers integration-->
```
The pom contains section of 
developers->developer

Please add yourself there with something like : ALM Octane Starteam support

radislavB commented on src/main/java/com/microfocus/application/automation/tools/octane/model/processors/scm/StarTeamSCMProcessor.java:
```diff
@@ -0,0 +1,252 @@
+        logger.warn("failed to obtain BuildData; no SCM repository info will be available");
+      }
+    } else {
+      logger.warn("failed to obtain BuildData; no SCM repository info will be available");
```
please change the message so we will understand what happen , for example "starTeamData is received null, no SCM repository info will be available"

radislavB commented on src/main/java/com/microfocus/application/automation/tools/octane/model/processors/scm/StarTeamSCMProcessor.java:
```diff
@@ -0,0 +1,252 @@
+        String branch = getSCMRepositoryBranch(starTeamData);
+        result = dtoFactory.newDTO(SCMRepository.class).setType(SCMType.STARTEAM).setUrl(url).setBranch(branch);
+      } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException | SecurityException e) {
+        logger.warn("failed to obtain BuildData; no SCM repository info will be available");
```
please add to warn exception class name (e.getClass().getSimpleName())+ exception message

radislavB reviewed (changes requested):
I have small comments.
There is a test that failing **_testSCMType(com.microfocus.application.automation.tools.octane.EnumsTest)_** - please fix it
If you need help with validation - please contact me.

paul-murray-mf committed:
Fix pull request #228 review comments

Fix EnumsTest
Add details to developers section
Improve warning messages with class name and more context

radislavB commented on pom.xml:
```diff
@@ -48,7 +48,7 @@
 		<msbuild.exe>C:\Windows\Microsoft.NET\Framework\v4.0.30319\msbuild.exe</msbuild.exe>
 		<msbuild.configuration>Release</msbuild.configuration>
-		<jenkins.version>2.121.3</jenkins.version>
+		<jenkins.version>2.138.4</jenkins.version>
```
Please revert jenkins version change. Seems like tests are failing because of this.

radislavB commented on src/main/java/com/microfocus/application/automation/tools/octane/model/processors/scm/StarTeamSCMProcessor.java:
```diff
@@ -0,0 +1,252 @@
+        String branch = getSCMRepositoryBranch(starTeamSCM);
+        result = dtoFactory.newDTO(SCMRepository.class).setType(SCMType.STARTEAM).setUrl(url).setBranch(branch);
+      } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException | SecurityException e) {
+        logger.warn(e.getClass().getSimpleName() + " unable to extract data from starTeamSCM, no SCM repository info will be available.");
```
please add here exception message

paul-murray-mf committed:
Address comments on PR 228

Revert Jenkins version to 2.121.3
Add exception to warning
